src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_container_emitter.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "jit_container_emitter.hpp"
+
+using namespace Xbyak_aarch64;
+
+namespace ov {
+namespace intel_cpu {
+namespace aarch64 {
+
+using jit_generator = dnnl::impl::cpu::aarch64::jit_generator;
+using cpu_isa_t = dnnl::impl::cpu::aarch64::cpu_isa_t;
+
+jit_container_emitter::jit_container_emitter(dnnl::impl::cpu::aarch64::jit_generator* h, dnnl::impl::cpu::aarch64::cpu_isa_t isa) : jit_emitter(h, isa) {
+    in_out_type_ = emitter_in_out_map::gpr_to_gpr;
+}
+
+void jit_container_emitter::map_abstract_registers(mapping_info& gpr_map_pool, mapping_info& vec_map_pool,
+                                                   snippets::lowered::LinearIR::container& expressions) const {
+    OV_CPU_JIT_EMITTER_ASSERT(!expressions.empty(), "Cannot map registers when there is no allocated_emitters provided");
+
+    auto map_regs = [&](const std::vector<snippets::Reg>& abstract_regs) {
+        std::vector<snippets::Reg> physical_regs = abstract_regs;
+        for (size_t i = 0; i < abstract_regs.size(); ++i) {
+            const auto& abstract_reg = abstract_regs[i];
+            const auto& type = abstract_reg.type;
+            const auto& abstract = abstract_reg.idx;
+            OV_CPU_JIT_EMITTER_ASSERT(one_of(type, snippets::RegType::gpr, snippets::RegType::vec), "Incorrect reg type detected!");
+            auto& mapping = type == snippets::RegType::gpr ? gpr_map_pool : vec_map_pool;
+            auto& abstract_to_physical = mapping.first;
+            auto& regs_pool = mapping.second;
+            auto& physical = physical_regs[i];
+            if (abstract_to_physical.count(abstract) == 0) {
+                OV_CPU_JIT_EMITTER_ASSERT(!regs_pool.empty(), "Cannot map registers for jit_container_emitter: not enough regs in the pool");
+                physical.idx = regs_pool.back();
+                regs_pool.pop_back();
+                abstract_to_physical[abstract] = physical.idx;
+            } else {
+                physical.idx = abstract_to_physical[abstract];
+            }
+        }
+        return physical_regs;
+    };
+
+    for (const auto& expression : expressions) {
+        std::vector<snippets::Reg> in_physical_regs, out_physical_regs;
+        std::vector<snippets::Reg> in_abstract_regs, out_abstract_regs;
+        std::tie(in_abstract_regs, out_abstract_regs) = expression->get_reg_info();
+        in_physical_regs = map_regs(in_abstract_regs);
+        out_physical_regs = map_regs(out_abstract_regs);
+        expression->set_reg_info({in_physical_regs, out_physical_regs});
+        if (auto container = std::dynamic_pointer_cast<jit_container_emitter>(expression->get_emitter()))
+            container->map_abstract_registers(gpr_map_pool, vec_map_pool, expressions);
+    }
```

Comment:
Is it the same as for `x64`? I's minor, but: can we extract and reuse?

Reply:
Thanks for the good idea!
I think we can only implement interface class without reference to arch and derive `jit_kernel_emitter` from the corresponding `jit_emitter` and `IContainer`.

@IvanNovoselov what do you think as author of this class? Can we create common for archs interface class with single method?

Reply:
@a-sidorova This register mapping was implemented in the jit_container_emitter class way before LinearIR was created, because at that time we had no convenient way to represent emitter sequence. Now we have it, so this mapping should be rewritten as a normal LIR pass (just like AssignRegisters). When this is done, I believe that the `jit_container_emitter` class could be deprecated.

I do not insist on doing it in this PR though. As an intermediate solution, I agree that we can leave `jit_container_emitter` as a stand-alone class (arch independent), and derive arch-specific jit_emitters that need mapping from it. 
Not sure why do you call it an interface class though, it will  be just a class with one method - `map_abstract_registers`

Reply:
Applied as https://github.com/openvinotoolkit/openvino/pull/23180#discussion_r1527955547. Thanks Alexandra, Ivan and tatyana-oliva!